Sources/SwiftInspectorCore/Tests/PropertyAnalzyerSpec.swift

```diff
@@ -87,11 +87,55 @@ final class PropertyAnalyzerSpec: QuickSpec {
               name: "thing",
               typeAnnotation: "String",
               comment: "",
-              modifier: [.public, .instance])
+              modifiers: [.public, .instance])
           ]
         }
       }
 
+      context("when there is a property in a nested type with the same type name") {
+        beforeEach {
+          let content = """
+          public final class FakeType {
+            public var thing: String = "Hello, World"
+
+            enum FakeType {
+              static let foo: String = "Hola"
+            }
+          }
+          """
+          fileURL = try? Temporary.makeFile(content: content)
+        }
+
+        it("detects the type name") {
+          let result = try? sut.analyze(fileURL: fileURL)
+          expect(result?.name) == "FakeType"
+        }
+
+        /*
+         This is actually not the ideal result of this and is a limitation of the implementation
+         Ideally you would have to pass in `FakeType.FakeType` to get this nested type's property
+         information. For now we are accepting this limitation and have this test to showcase
+         what happens in this scenario.
```

Comment:
Perfect 👌

Reply:
I increasingly feel like unit tests are a great way to document edge cases and accepted limitations.